### PR TITLE
Hyperlink

### PR DESCRIPTION
**Pull request by sahajp23**
# Author Checklist
1. Changes address original issue? No original issue to address
2. Unit tests included and/or updated with changes? Yes
3. Command line build passes?  Yes
4. Changes have been smoke-tested? Yes

Adds a new Domain Object type, a Hyperlink, which can be added as either a button or a link that may be opened in this window or another depending on user preference. Modified view in display layout to remove the header and title for the hyperlink domain object. 
@larkin @prestonjcrowe @luisschubert @adoubekk Would be great to get a set of eyes on this

### CODE REVIEW

File: platform/features/listview/test/controllers/ListViewControllerSpec.js [hunk 1]
```diff
@@ -86,8 +86,6 @@ define(
                 domainObject.getCapability.andReturn(
                     mutationCapability
                 );
-
-
                 scope = jasmine.createSpyObj(
```
Comment (luisschubert) [inline platform/features/listview/test/controllers/ListViewControllerSpec.js]: this seems like an unrelated/unnecessary change.

File: platform/features/hyperlink/test/HyperlinkControllerSpec.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+define(
```
Comment (luisschubert) [inline platform/features/hyperlink/test/HyperlinkControllerSpec.js]: Make sure that all your files contain the license at the top.

File: platform/features/hyperlink/bundle.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                    "required": true,
+                    "cssClass": "l-input-lg"
+                  },
+                  {
+                    "key": "showTitle",
+                    "value": "false",
+                  },{
+                      "key": "displayFormat",
+                      "name": "Display Format",
+                      "control": "composite",
```
Comment (larkin) [inline platform/features/hyperlink/bundle.js]: No reason to wrap the select control in a composite control; this should just be a select control.

File: platform/features/hyperlink/bundle.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+                                  }
+                              ],
+                              "cssClass": "l-inline"
+                          }
+                      ]
+                  },
+                  {
+                      "key": "openNewTab",
+                      "name": "Tab to Open Hyperlink",
+                      "control": "composite",
```
Comment (larkin) [inline platform/features/hyperlink/bundle.js]: No reason to wrap the select control in a composite control; this should just be a select control.

File: platform/features/hyperlink/bundle.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+                              ],
+                              "cssClass": "l-inline"
+                          }
+                      ]
+                  }
+                ],
+                "model": {
+                    "displayFormat": ["link"],
+                    "openNewTab": ["thisTab"],
+                    "showTitle":["false"]
```
Comment (larkin) [inline platform/features/hyperlink/bundle.js]: Should be `"showTitle": true`; not an array.

File: platform/features/hyperlink/bundle.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+                }
+
+              }
+        ],
+        "views": [
+            {
+                "key": "example.hyperlink",
```
Comment (larkin) [inline platform/features/hyperlink/bundle.js]: should not include "example"

File: platform/features/hyperlink/bundle.js [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+                "type": "example.hyperlink",
```
Comment (larkin) [inline platform/features/hyperlink/bundle.js]: should not include "example"

File: platform/features/hyperlink/bundle.js [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+                "cssClass": "icon-check",
+                "name": "List",
```
Comment (larkin) [inline platform/features/hyperlink/bundle.js]: Name shouldn't be List.

File: platform/features/hyperlink/res/templates/hyperlink.html [hunk 9]
```diff
@@ -0,0 +1,20 @@
+<div class="s-hyperlink" ng-controller="HyperlinkController as hyperlink">
+      <div ng-if="hyperlink.isButton()">
+          <div ng-if="hyperlink.openNewTab()">
+              <a class="s-button"
+              href ={{domainObject.getModel().url}} target="_blank">{{domainObject.getModel().displayText}}</a>
+          </div>
+          <div ng-if="hyperlink.openNewTab()== false">
```
Comment (larkin) [inline platform/features/hyperlink/res/templates/hyperlink.html]: Instead of nested ifs here, you can combine this into a single element and use more specific directives:
* `ng-class` will allow you to add a class to a element when something is true
* `ng-attr-target` will allow you to conditionally set the target.

For example, this could be:
```
<div >
    <a href="{{domainObject.getModel().url}}" 
       ng-attr-target="{{hyperlink.openNewTab() ? '_blank' : undefined}}"
       ng-class="{
           's-button': hyperlink.isButton()
       }"
       >
        {{domainObject.getModel().displayText}}
    </a>
</div>
```

File: platform/features/layout/res/templates/frame.html [hunk 10]
```diff
@@ -22,11 +22,13 @@
 <div class="frame frame-template abs">
     <div class="abs object-browse-bar l-flex-row">
         <div class="left flex-elem l-flex-row grows">
-            <mct-representation
-                key="'object-header'"
-                mct-object="domainObject"
-                class="l-flex-row flex-elem object-header grows">
-            </mct-representation>
+            <div ng-if=domainObject.getModel().showTitle!="false">
```
Comment (larkin) [inline platform/features/layout/res/templates/frame.html]: Should be `ng-if="domainObject.getModel().showTitle"` when it's a boolean.

File: platform/features/layout/res/templates/frame.html [hunk 11]
```diff
@@ -35,9 +37,14 @@
             </mct-representation>
         </div>
     </div>
-    <div class="abs object-holder">
-        <mct-representation key="representation.selected.key"
-                            mct-object="representation.selected.key && domainObject">
-        </mct-representation>
+    <div ng-if=domainObject.getModel().showTitle=="false" class="l-flex-row flex-elem object-header grows">
```
Comment (larkin) [inline platform/features/layout/res/templates/frame.html]: Should be `ng-if="!domainObject.getModel().showTitle"` when it's a boolean.

File: platform/features/layout/res/templates/frame.html [hunk 12]
```diff
@@ -35,9 +37,14 @@
...
+          <mct-representation key="representation.selected.key"
+                              mct-object="representation.selected.key && domainObject">
+          </mct-representation>
+    </div>
+    <div ng-if=domainObject.getModel().showTitle!="false" class="abs object-holder">
```
Comment (larkin) [inline platform/features/layout/res/templates/frame.html]: Should be `ng-if="domainObject.getModel().showTitle"` when it's a boolean.

File: platform/features/listview/test/controllers/ListViewControllerSpec.js (hunk 1 above)
Comment (larkin) [inline platform/features/listview/test/controllers/ListViewControllerSpec.js]: yeah shouldn't change a file that's not in scope of your pull request.

File: platform/features/hyperlink/test/HyperlinkControllerSpec.js [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+                "getModel",
+                ["link"]
+              );
+              domainObject.getModel.link= link
+            });
+            it("opens a specific given link", function () {
+                expect(domainObject.getModel.link)
+                    .toEqual("http://nasa.gov");
+            });
+
```
Comment (larkin) [inline platform/features/hyperlink/test/HyperlinkControllerSpec.js]: Test the methods you defined on the controller, e.g.  "it knows when it should open in new tab" and "it knows when it is a button".

**Comment by larkin** [comment]
Closing as now tracked via new PR: https://github.com/nasa/openmct/pull/1660/files
